armory/utils/metrics.py

```diff
@@ -2011,6 +2098,13 @@ def results(self):
                         raise ZeroDivisionError(
                             f"No values to calculate WER in {prefix}_{metric.name}"
                         )
+                if metric.name == "entailment":
```

Comment:
While this conditional block maps integers to strings, in the `if self.full:` code block above, the integer is added to the results dict. 

So you get an output JSON like such:
```
"benign_entailment": [
            2
        ],
 "benign_total_entailment": {
            "entailment": 1,
            "total": 1
        },
```
The integers don't mean anything to a user right? So should they be omitted from the results JSON?

Reply:
You also get logs as such:

```
2022-04-19 17:29:59 1m21s SUCCESS  armory.utils.metrics:log_task:2054 Average entailment on adversarial test examples relative to ground truth labels: 200.00%
```

It appears confusing to me but again I might be lacking context. Is that the desired behavior?

Reply:
Yeah, a lot of this has to do with the brittleness of the current MetricsLogger system. I will try making something a little nicer.

Reply:
This update should be a lot better.